gammapy/detect/cwt.py

```diff
@@ -57,200 +60,390 @@ def difference_of_gauss_kernel(radius, scale_step, n_sigmas=8):
 
 
 class CWT(object):
-    """Continuous wavelet transform.
+    """
+    Continuous wavelet transform.
 
-    TODO: describe algorithm
+    TODO: describe algorithm (modify the words below)
 
-    TODO: instead of storing all the arrays as data members,
-    we could have a ``.data`` member which is a dict of images?
+    Depending on their spectral index, sources won't have the same characteristic scale.
+    Therefore to detect sources, we need to compute the wavelet transform at several
+    scales in order to search for various PSF sizes. Then for each scale, the wavelet
+    transform values under the given significance threshold are rejected. This gives us
+    a multiscale support. Then, using the reconstruction by continuous wavelet packets,
+    we obtain a filtered image yielding the detected sources.
+    To compute the threshold map for a given scale a, the standard EGRET diffuse
+    background model to which was added the flux of the extragalactic background,
+    and the exposure for the considered energy range were used.
 
-    TODO: give references
+    Parameters
+    ----------
+    kernels : '~gammapy.detect.CWTKernels'
+        Kernels for the algorithm.
+    max_iter : int, optional (default 10)
+        The maximum number of iterations of the CWT algorithm.
+    tol : double, optional (default 1e-5)
+        Tolerance for stopping criterion.
```

Comment:
You should add an explanation what the stopping criterion is.
A sentence or formula or something like that. Maybe as part of the main docstring text, or in a `Notes` section below?